### PR TITLE
nrsc5: init at 20171129

### PR DESCRIPTION
**Pull request by markuskowa**
###### Motivation for this change
NRSC5 is a comand line tool to receive HD-Radio (digital radio in North America) with
a RTL-SDR radio device. 

###### Things done
Tested with a RTL-SDR USB stick.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/faad2/faad2-hdc-support.patch [hunk 1]
```diff
@@ -0,0 +1,644 @@
+From 176e496b2a5461af122e20054079887b0b3d46a8 Mon Sep 17 00:00:00 2001
```
Comment (Mic92) [inline pkgs/development/libraries/faad2/faad2-hdc-support.patch]: Is this an upstream patch to the project?

Comment (Mic92) [reply pkgs/development/libraries/faad2/faad2-hdc-support.patch]: Ah, so this feature is basically derived from this fork? https://github.com/dsvensson/faad2

File: pkgs/development/libraries/faad2/default.nix [hunk 2]
```diff
@@ -12,8 +13,11 @@ stdenv.mkDerivation rec {
     sha256 = "1db37ydb6mxhshbayvirm5vz6j361bjim4nkpwjyhmy4ddfinmhl";
   };
 
+  patches = [ optional hdcSupport ./faad2-hdc-support.patch ];
```
Comment (grahamc) [inline pkgs/development/libraries/faad2/default.nix]: This is making an array of three things:

1. the `optional` function
2. the value of `hdcSupport`
3. the patch

To make it do what you expect, you'll need:

```
patches = [ (optional hdcSupport ./faad2-hdc-support.patch) ];
```

File: pkgs/development/libraries/faad2/faad2-hdc-support.patch (hunk 1 above)
Comment (markuskowa) [reply pkgs/development/libraries/faad2/faad2-hdc-support.patch]: The patch is from the nrsc5 package itself, which pulls from https://github.com/dsvensson/faad2
during the build process and applies the patch. An alternative solution would be
to add a separate package faad2-hdc, which is build from dsvensson faad2 branch.

File: pkgs/development/libraries/faad2/default.nix (hunk 2 above)
Comment (markuskowa) [reply pkgs/development/libraries/faad2/default.nix]: Thanks for the hint

**Comment by markuskowa** [comment]
Thanks for your feedback. I have rewrote the expression. The new solution is now self contained,
less invasive, and does not create an unnecessary cross dependence to the faad2 package.

**Pull request merged**